### PR TITLE
Refactored code

### PR DESCRIPTION
**Pull request by pbywater**
- Added watchify
- Added tests (some need to be fixed now)
- Made functions testable/ purer
- Nodes and links now appear on page but draggable function isn't fully working yet
- Added comments/removed commented out code

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: src/client/helpers/helpers.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// filters array for unique values
+function onlyUnique(value, index, self) {
+  return self.indexOf(value) === index;
+}
+
+// generic function to find radial x and y using distance and angle
+function calculateXY(nodeList) {
```
Comment (des-des) [inline src/client/helpers/helpers.js]: Would be better to map the nodes then foreach + mutate

File: src/client/helpers/helpers.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+// filters array for unique values
+function onlyUnique(value, index, self) {
```
Comment (des-des) [inline src/client/helpers/helpers.js]: I think you are not using this function (:fire:)

File: src/client/node_transformations.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    rest: nodesArrayCopy.slice(1),
+  };
+};
+
+const tagNodesByTag = (tagsArray, startingCx, startingCy, generateId) => {
+  const out = {};
+  const arrayCopy = tagsArray.slice(0);
+  arrayCopy.forEach((tag, i) => {
+    const newId = generateId();
+    out[tag.max.tag] = {
```
Comment (des-des) [inline src/client/node_transformations.js:22]: tag === tag.max.tag?

File: src/client/node_transformations.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+const getXAndY = (angle, distance, startingCx, startingCy) => {
+  const nx = startingCx + Math.cos(angle) * distance;
+  const ny = startingCy + Math.sin(angle) * distance;
+  return {
+    x: nx,
+    y: ny,
+  };
+};
+
+const getMemoryNodePositions = (tagNode, numTagMemories, memoryIndex, currentAvgRating) => {
```
Comment (des-des) [inline src/client/node_transformations.js:43]: :+1:

File: src/client/node_transformations.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+const memoryNodesAndLinks = (tagNodes, memoriesByTag) => {
+  const nodes = {};
+  const links = [];
+
+  Object.keys(tagNodes).forEach((tag, index) => {
+    const tagNode = tagNodes[tag];
+    const tagMemoriesRest = memoriesByTag[index].rest;
+    const numMemoriesRest = memoriesByTag[index].rest.length;
+    nodes[tagNodes[tag].id] = tagNodes[tag];
+    const memoryNodes = tagMemoriesRest.map((tagMemory, memoryIndex) => {
```
Comment (des-des) [inline src/client/node_transformations.js]: Remove `const memoryNodes = ` variable called `memoryNodes` not being used
`tagMemoriesRest.map((tagMemory, memoryIndex) => {`

File: tests/src/client/node_transformations.js [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+          likes: 18,
+          tag: "pets",
+        },
+      ]
+    }
+  ];
+  const startingCx = 160;
+  const startingCy = 120;
+
+  const expected = { family: { avgRating: 9.2, id: 0, likes: 19, tag: 'family', x: 160, y: 120 }, friends: { avgRating: 9.2, id: 1, likes: 19, tag: 'friends', x: 160, y: 320 }, pets: { avgRating: 9.2, id: 2, likes: 19, tag: 'pets', x: 160, y: 520 } };
```
Comment (des-des) [inline tests/src/client/node_transformations.js]: Break this over multiple lines!

File: tests/src/client/node_transformations.js [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+    tagNodesByTag(arrayCopy, startingCx, startingCy, generateId),
+    expected,
+    'creates map of tag nodes, with ids and positions'
+  );
+
+  t.end();
+});
+
+test('memoryNodesAndLinks', (t) => {
+  const memoriesByTag = { family: { avgRating: 9.2, id: 0, likes: 19, tag: 'family', x: 160, y: 120 }, friends: { avgRating: 9.2, id: 1, likes: 19, tag: 'friends', x: 160, y: 320 }, pets: { avgRating: 9.2, id: 2, likes: 19, tag: 'pets', x: 160, y: 520 } };
```
Comment (des-des) [inline tests/src/client/node_transformations.js]: And this

**Review by des-des — APPROVED** [review]

**Review by des-des — CHANGES_REQUESTED** [review]
ops wrong button

**Comment by des-des** [comment]
:tada:

File: src/client/node_transformations.js (hunk 3 above)
Comment (pbywater) [reply src/client/node_transformations.js:22]: That's confusing.. But I think right! Not sure if there's a better way of phrasing it

**Comment by pbywater** [comment]
@des-des All done I think!

**Comment by des-des** [comment]
delete calculateXY and will merge

**Comment by pbywater** [comment]
@des-des deleted! :)

**Review by des-des — APPROVED** [review]

**Pull request merged**